src/fft/polynomial.rs

```diff
@@ -192,6 +206,7 @@ impl<'a, 'b> Add<&'a Polynomial> for &'b Polynomial {
             }
             result
         };
+        // If the leading coefficients end up being zero, pop them off.
         result.truncate_leading_zeros();
```

Comment:
just noticed that this will never be reached, can we delete these two lines?

Reply:
I had to check myself, but we are indeed reaching these lines. The if-else returns a result which is stored in the mutable `result` variable, which is then being truncated at the end.

Reply:
Ohh true! I missed that part!